src/Metadata/Property/Factory/AnnotationPropertyCollectionMetadataFactory.php

```diff
@@ -22,7 +22,7 @@
  *
  * @author Kévin Dunglas <dunglas@gmail.com>
  */
-final class AnnotationPropertyNameCollectionFactory implements PropertyNameCollectionFactoryInterface
+final class AnnotationPropertyCollectionMetadataFactory implements PropertyNameCollectionFactoryInterface
```

Comment:
The problematic class that I noticed earlier was actually `ApiPlatform\Core\Metadata\Resource\Factory\AnnotationResourceMetadataFactory`.

It even has some invalid use statements... (I'm not sure how it's functional, or do those code paths not get executed at all?)